augur/curate/__init__.py

```diff
@@ -0,0 +1,53 @@
+"""
+A suite of commands to help with data curation.
+"""
+import sys
+
+from augur.argparse_ import add_command_subparsers
+from augur.errors import AugurError
+from augur.io.json import dump_ndjson, load_ndjson
+from . import normalize_strings
+
+
+SUBCOMMAND_ATTRIBUTE = '_curate_subcommand'
+SUBCOMMANDS = [
+    normalize_strings,
+]
+
+
+def register_parser(parent_subparsers):
+    shared_parser = create_shared_parser()
+    parser = parent_subparsers.add_parser("curate",
+        help=__doc__)
+
+    # Add print_help so we can run it when no subcommands are called
+    parser.set_defaults(print_help = parser.print_help)
+
+    # Add subparsers for subcommands
+    subparsers = parser.add_subparsers(dest="subcommand", required=False)
+    # Using a subcommand attribute so subcommands are not directly
+    # run by top level Augur. Process I/O in `curate`` so individual
+    # subcommands do not have to worry about it.
+    add_command_subparsers(subparsers, SUBCOMMANDS, SUBCOMMAND_ATTRIBUTE)
+
+    return parser
+
+
+def run(args):
+    # Print help if no subcommands are used
+    if not getattr(args, SUBCOMMAND_ATTRIBUTE, None):
+        return args.print_help()
+
+    # Read inputs
+    if not sys.stdin.isatty():
+        records = load_ndjson(sys.stdin)
+    else:
+        raise AugurError("No valid inputs were provided.")
+
+    # Add input records to the argparse Namespace so they are available for subcommands
+    args._records = records
+    # Run subcommand to get modified records
+    modified_records = getattr(args, SUBCOMMAND_ATTRIBUTE).run(args)
```

Comment:
Yeah, not sure what I was thinking when I did this weird indirection.
Fixed up in cdacc376efd3a6d5299d26eb7a911ebfe2819d9b

Reply:
:raised_hands: I think of this sort of thing as an "incomplete thought" or "incomplete sentence". Like, when I do this, it's often because I've had a thought and started down that way with code/design to the point where it works, but then had another thought that makes it weird/indirect and never really got back to finishing thinking thru the original code/design.